ui/src/types/provider.d.ts

```diff
@@ -0,0 +1,18 @@
+export interface IdentityProvider {
```

Comment:
nitpick: these fields are all non-mandatory right now, according to the spec, so you would probably want to put them as optional

Reply:
Good point, thanks :)
Will change it.